plugins/scale/scale.cpp

```diff
@@ -691,32 +692,59 @@ class wayfire_scale : public wf::plugin_interface_t
         output->render->damage_whole();
     }
 
-    /* Returns a list of views for all workspaces */
+    /* Returns a list of views for all workspaces
+     * Also takes minimized views out of the minimized layer
+     * and stores the state */
     std::vector<wayfire_view> get_all_workspace_views()
     {
         std::vector<wayfire_view> views;
 
+        uint32_t layer_mask = wf::LAYER_WORKSPACE;
+        if (show_minimized)
+        {
+            layer_mask |= wf::LAYER_MINIMIZED;
+        }
+
         for (auto& view :
-             output->workspace->get_views_in_layer(wf::LAYER_WORKSPACE))
+             output->workspace->get_views_in_layer(layer_mask))
```

Comment:
It would have been possible to query all views in get_views() but that would require an additional for loop which I think is unnecessary